### ISSUE TITLE
feat(core,platform): rxjs debounceTime instead of delay

### ISSUE THREAD

**Issue by platon-rov** [SAP/fundamental-ngx#7911]
#### Is this a bug, enhancement, or feature request?

Enhancement.

#### Briefly describe your proposal.

We very often use rxjs' `delay` to postpone some processing and improve the performance. But, unfortunately, `delay` only delays the processing and doesn't decrease amount of processing. To enhance the performance `debounceTime` should be used instead.

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

Latest on 25th of March, 2022.

#### Is there anything else we should know?

> Emits a notification from the source Observable only after a particular time span has passed without another source emission.
https://rxjs.dev/api/operators/debounceTime

> Delays the emission of items from the source Observable by a given timeout or until a given Date.
https://rxjs.dev/api/operators/delay
